samcli/commands/local/invoke/core/options.py

```diff
@@ -0,0 +1,85 @@
+"""
+Invoke Command Options related Datastructures for formatting.
+"""
+from typing import Dict, List
+
+from samcli.cli.row_modifiers import RowDefinition
+
+# NOTE(sriram-mv): The ordering of the option lists matter, they are the order
+# in which options will be displayed.
+
+REQUIRED_OPTIONS: List[str] = ["template_file"]
+
+AWS_CREDENTIAL_OPTION_NAMES: List[str] = ["region", "profile"]
+
+TEMPLATE_OPTIONS: List[str] = [
+    "parameter_overrides",
+]
+
+CONTAINER_OPTION_NAMES: List[str] = [
+    "event",
+    "no_event",
+    "env_vars",
+    "container_env_vars",
+    "debug_port",
+    "debugger_path",
+    "debug_args",
+    "docker_volume_basedir",
+    "skip_pull_image",
+    "docker_network",
+    "force_image_build",
+    "shutdown",
+    "container_host",
+    "container_host_interface",
+    "invoke_image",
+]
+
+BETA_OPTION_NAMES: List[str] = ["beta_features"]
+
+CONFIGURATION_OPTION_NAMES: List[str] = ["config_env", "config_file"]
+
+EXTENSION_OPTIONS: List[str] = ["hook_name", "skip_prepare_infra"]
+
+ARTIFACT_LOCATION_OPTIONS: List[str] = [
+    "log_file",
+    "layer_cache_basedir",
+]
+
+OTHER_OPTIONS: List[str] = ["debug"]
+
+ALL_OPTIONS: List[str] = (
+    REQUIRED_OPTIONS
+    + TEMPLATE_OPTIONS
+    + AWS_CREDENTIAL_OPTION_NAMES
+    + CONTAINER_OPTION_NAMES
+    + BETA_OPTION_NAMES
+    + ARTIFACT_LOCATION_OPTIONS
+    + EXTENSION_OPTIONS
+    + CONFIGURATION_OPTION_NAMES
+    + OTHER_OPTIONS
+)
+
+OPTIONS_INFO: Dict[str, Dict] = {
+    "Required Options": {"option_names": {opt: {"rank": idx} for idx, opt in enumerate(REQUIRED_OPTIONS)}},
+    "Template Options": {"option_names": {opt: {"rank": idx} for idx, opt in enumerate(TEMPLATE_OPTIONS)}},
+    "AWS Credential Options": {
+        "option_names": {opt: {"rank": idx} for idx, opt in enumerate(AWS_CREDENTIAL_OPTION_NAMES)}
+    },
+    "Container Options": {"option_names": {opt: {"rank": idx} for idx, opt in enumerate(CONTAINER_OPTION_NAMES)}},
+    "Beta Options": {"option_names": {opt: {"rank": idx} for idx, opt in enumerate(BETA_OPTION_NAMES)}},
```

Comment:
Does "Beta Options" has to be before "Artifact Location Options", "Extension Options" and "Configuration Options"? 

Asking because in another PR #5062, we intend to move "Beta Options" to part of the common options along with "Other Options". And they will be always at the bottom.

Reply:
It's in that place because the beta options were closely tied to container options at that point. I do not have concerns moving it down. Should we then indicate with UX markings that a particular command has beta options up top somewhere before listing the actual options?

My concern is that users do not see them if we move them down.

Reply:
The fact that `OPTIONS_INFO` is a dict guarantees the keys' iteration order are always the same as the order they were inserted. So if we have "Beta Options" here at an upper position, the `add_common_options_info` function in the other PR won't change the ordering. We can still keep "Beta Options" as it is if we prefer it at an upper position.

> Should we then indicate with UX markings that a particular command has beta options up top somewhere before listing the actual options?
Not sure what it looks like, but open to it.

> My concern is that users do not see them if we move them down.
IMO it's easier for users to see if it's at bottom, because you usually see the bottom of previous command run when you type the new command. But I guess what we want is a consensus. I personally don't have a strong preference.